File: imc-infrastructure/src/main/java/com/intel/missioncontrol/persistence/insight/primitives/ProcessSettings.java
/**
 * Copyright (c) 2020 Intel Corporation
 *
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package com.intel.missioncontrol.persistence.insight.primitives;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class ProcessSettings {
    @SerializedName("tools_set")
    @Expose
    @Nullable
    private String toolsSet;

    @SerializedName("mapType")
    @Expose
    @Nullable
    private String mapType;

    @SerializedName("analytics")
    @Expose
    @NotNull
    private List analytics = new ArrayList();

    @SerializedName("inspection")
    @Expose
    @Nullable
    private Inspection inspection;

    @SerializedName("mapping")
    @Expose
    @Nullable
    private Mapping mapping;

    @Nullable
    public final String getToolsSet() {
        return this.toolsSet;
    }

    public final void setToolsSet(@Nullable String var1) {
        this.toolsSet = var1;
    }

    @Nullable
    public final String getMapType() {
        return this.mapType;
    }

    public final void setMapType(@Nullable String var1) {
        this.mapType = var1;
    }

    @NotNull
    public final List getAnalytics() {
        return this.analytics;
    }

    public final void setAnalytics(@NotNull List var1) {
        this.analytics = var1;
    }

    @Nullable
    public final Inspection getInspection() {
        return this.inspection;
    }

    public final void setInspection(@Nullable Inspection var1) {
        this.inspection = var1;
    }

    @Nullable
    public final Mapping getMapping() {
        return this.mapping;
    }

    public final void setMapping(@Nullable Mapping var1) {
        this.mapping = var1;
    }

}

File: LibInsight/src/main/kotlin/com/intel/insight/datastructures/ProcessSettings.kt
/**
 * Copyright (c) 2020 Intel Corporation
 *
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package com.intel.insight.datastructures

import java.util.ArrayList
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class ProcessSettings {
    @SerializedName("tools_set")
    @Expose
    var toolsSet: String? = null
    @SerializedName("mapType")
    @Expose
    var mapType: String? = null
    @SerializedName("analytics")
    @Expose
    var analytics: List<Any> = ArrayList()
    @SerializedName("inspection")
    @Expose
    var inspection: Inspection? = null
    @SerializedName("mapping")
    @Expose
    var mapping: Mapping? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ProcessSettings

        if (mapType != other.mapType) return false
        if (analytics != other.analytics) return false
        if (inspection != other.inspection) return false

        return true
    }

    override fun hashCode(): Int {
        var result = mapType?.hashCode() ?: 0
        result = 31 * result + analytics.hashCode()
        result = 31 * result + (inspection?.hashCode() ?: 0)
        return result
    }

}
